learning_journal/views.py

```diff
@@ -0,0 +1,72 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from .form import JournalForm
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy import desc
+import transaction
+import markdown
+
+from .models import (
+    DBSession,
+    Entry,
+    )
+
+
+@view_config(route_name='home', renderer='templates/list_view.jinja2')
+def list_view(request):
+    """Handle the view of our home page."""
+    return {'entries': DBSession.query(Entry).order_by(desc(Entry.created)).all()}
+
+
+@view_config(route_name='detail_view', renderer='templates/detail_view.jinja2')
+def detail_view(request):
+    """Handle the view of a single journaly entry."""
+    md = markdown.Markdown(safe_mode='replace', html_replacement_text='NO')
+    this_id = request.matchdict['this_id']
+    entry = DBSession.query(Entry).get(this_id)
+    text = md.convert(entry.text)
+    return {'entry': entry, 'text': text}
+
+
+@view_config(route_name='add_view', renderer='templates/add_view.jinja2')
+def add_view(request):
+    """Handle the view of our adding new entry page."""
+    form = JournalForm(request.POST)
+    if request.method == "POST" and form.validate():
+        new_entry = Entry(title=form.title.data, text=form.text.data)
+        DBSession.add(new_entry)
+        DBSession.flush()
+        this_id = new_entry.id
+        return HTTPFound(location='/detail/{}'.format(this_id))
```

Comment:
You can (and should) use the `request.route_url` method here to build the correct url for your new location.  This approach is dangerous and fragile.